### PR TITLE
add documentation for `key` attribute on components

### PR DESCRIPTION
**Pull request by pedramphp**
I have added an example with the documentation. feel free to revise it.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated/added documentation affected by my changes.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -334,6 +334,46 @@ The containing component can reference the repeated DOM elements using the follo
...
+   }
+   
+   changeChildButtonColor() {
+   	this.btnChild.changeColor('red');
+   }
+}
+
+<div.btnContainer on-click=('changeChildButtonColor')>
+   <custom-button key='btn-child' />
+</div>
```
Comment (austinkelleher) [inline docs/components.md]: Also looks like you forgot to end your markdown code block here.

Comment (pedramphp) [inline docs/components.md]: fixed

**Comment by pedramphp** [comment]
We can try to update the main paragraph for the `key` to more content regarding the component usecase.
https://github.com/marko-js/marko/blob/master/docs/components.md#key

is this what you are referring to?

**Comment by austinkelleher** [comment]
@pedramphp Yes, that was what I was referring to. I don't think there should be two separate sections discussing the `key` attribute.

**Comment by mlrawlings** [comment]
This was addressed by Patrick in 0344faf8ab133aeddd9bce1c744341e6d85811b5

**Pull request closed without merging**